sos/report/plugins/microovn.py

```diff
@@ -0,0 +1,152 @@
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This is based on code from ovn_central.py plugin
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+import json
+import os
+from sos.report.plugins import Plugin, UbuntuPlugin
+
+
+class MicroOVN(Plugin, UbuntuPlugin):
+
+    short_desc = 'MicroOVN Snap'
+    plugin_name = "microovn"
+    profiles = ('network', 'virt')
+
+    packages = ('microovn', )
+
+    commands = ('microovn', 'microovn.ovn-appctl', 'microovn.ovn-nbctl',
+                'microovn.ovn-sbctl', 'microovn.ovn-trace',
+                'microovn.ovs-appctl',
+                'microovn.ovs-dpctl', 'microovn.ovs-ofctl',
+                'microovn.ovs-vsctl', 'microovn.ovsdb-client',
+                'microovn.ovsdb-tool',)
+
+    services = ('snap.microovn.central', 'snap.microovn.daemon',
+                'snap.microovn.ovn-ovsdb-server-nb',
+                'snap.microovn.refresh-expiring-certs',
+                'snap.microovn.switch', 'snap.microovn.chassis',
+                'snap.microovn.ovn-northd',
+                'snap.microovn.ovn-ovsdb-server-sb',
+                'snap.microovn.refresh-expiring-certs.timer',)
+
+    ovn_nbdb_sock_path = '/var/snap/microovn/common/run/ovn/ovnnb_db.ctl'
+    ovn_sbdb_sock_path = '/var/snap/microovn/common/run/ovn/ovnsb_db.ctl'
+    ovn_sock_path = '/var/snap/microovn/common/run/ovn'
+    ovn_controller_sock_regex = 'ovn-controller.*.ctl'
+    ovn_northd_sock_regex = 'ovn-northd.*.ctl'
+
+    def _find_sock(self, path, regex_name):
+        return os.path.join(path, regex_name)
+
+    def get_tables_from_schema(self, filename, skip=None):
+        """ Get tables from schema """
+        try:
+            fname = self.path_join(filename)
+            with open(fname, 'r', encoding='UTF-8') as file:
+                try:
+                    db_schema = json.load(file)
+                except Exception:  # pylint: disable=broad-except
+                    self._log_error(f"Cannot parse JSON file {filename}")
+                    return None
+        except IOError as ex:
+            self._log_error(
+                f"Could not open DB schema file {filename}: {ex}")
+            return None
+
+        try:
+            return [table for table in dict.keys(
+                db_schema['tables']) if table not in skip]
+        except AttributeError:
+            self._log_error(f"DB schema {filename} has no 'tables' key")
+        return None
+
+    def add_database_output(self, tables, cmds, ovn_cmd):
+        """ Collect OVN database output """
+        if not tables:
+            return
+        for table in tables:
+            cmds.append(f'{ovn_cmd} list {table}')
```

Comment:
The conditional check is redundant, since if `tables` has no elements, the for loop will do nothing and we'll fallthru to the implicit `None` return. However, rather than mutating the `cmds` list inside the method, this method should be returning an explicit list of formatted commands to add to the list which we can use with `extend()`, or we should just do the list comprehension directly, e.g. either:

```python
def add_database_output(self, tables, ovn_cmd):
return [f"ovn_cmd} list {table}" for table in tables]

[...]
 # Database Output
nb_tables = self.get_tables_from_schema(self.path_join('/snap/microovn/current/share/ovn', 'ovn-nb.ovsschema'))

cmds.extend(self.add_database_output(nb_tables, 'microovn.ovn-nbctl --no-leader-only'))
```

Or, doing it directly:

```python
nb_tables = self.get_tables_from_schema(self.path_join('/snap/microovn/current/share/ovn', 'ovn-nb.ovsschema'))

cmds.extend([
    f"{pfx}ovn-nbctl --no-leader-only list {table}" for table in nb_tables
])

[... then the same for nb_tables ..]
```

Reply:
I'm cleaning this up. This was directly taken from the ovn_central plugin, so it might be a good idea to replicate the clean-up over there too. I can send a separate PR for that if you want.